### ISSUE TITLE
How to customize scrolling shortcuts?

### ISSUE THREAD

**Issue by winklerrr** [vscode-neovim/vscode-neovim#1039]
Dear vscode-neovim team,

how can I customize the scrolling shortcuts, for example like <kbd>ctrl</kbd>+<kbd>u</kbd>?
What I want to do is to keep the cursor always in the center of the screen (via `z.` or `zz`), so I adapted my `keybindings.json`:

```json
[
  // disable the actual ctrl-u
  {
    "key": "ctrl+u",
    "command": "-vscode-neovim.ctrl-u",
    "when": "editorTextFocus && neovim.ctrlKeysNormal && neovim.init && neovim.mode != 'insert'"
  },

  // send the ctrl-u + z. for centering
  {
    "key": "ctrl+u",
    "command": "vscode-neovim.send",
    "when": "editorTextFocus && neovim.ctrlKeysNormal && neovim.init && neovim.mode != 'insert'",
    "args":  "<C-u>z."
  },
]
```

But after these changes, half-page up and down commands now jump 100 lines (instead of 20 like before). So I think something seems to be wrong here?

I also saw that there is some commented-out code in the `src/vim/vscode-scrolling.vim` file. Maybe this has something todo with the issue?

Related: https://stackoverflow.com/questions/73802761/vscode-neovim-how-to-jump-half-page-up-down-and-center-screen-vertically

**Comment by trkoch** [comment]
Have you tried to create a mapping in Vim?

**Comment by tom-pollak** [comment]
Scrolling dosen't work well for me at the moment, which I think will be fixed in #993 but a workaround I've found has been to use the [german-scroll extension](https://open-vsx.org/extension/Turiiya/german-scroll):

```json
  {
    "key": "ctrl+d",
    "command": "-vscode-neovim.ctrl-d",
    "when": "editorTextFocus && neovim.ctrlKeysNormal && neovim.init && neovim.mode != 'insert'"
  },
  {
    "key": "ctrl+d",
    "command": "-vscode-neovim.send",
    "when": "editorTextFocus && neovim.ctrlKeysNormal && neovim.init && neovim.mode != 'insert'"
  },
  {
    "key": "ctrl+d",
    "command": "germanScroll.bertholdDown",
    "when": "editorTextFocus && neovim.ctrlKeysNormal && neovim.init && neovim.mode != 'insert'",
  },
  {
    "key": "ctrl+d",
    "command": "-deleteRight",
    "when": "textInputFocus"
  },
  {
    "key": "ctrl+u",
    "command": "-vscode-neovim.ctrl-u",
    "when": "editorTextFocus && neovim.ctrlKeysNormal && neovim.init && neovim.mode != 'insert'"
  },
  {
    "key": "ctrl+u",
    "command": "-vscode-neovim.send",
    "when": "editorTextFocus && neovim.ctrlKeysNormal && neovim.init && neovim.mode != 'insert'"
  },
  {
    "key": "ctrl+u",
    "command": "germanScroll.bertholdUp",
    "when": "editorTextFocus && neovim.ctrlKeysNormal && neovim.init && neovim.mode != 'insert'",
  },
```

**Comment by theol0403** [comment]
This will be fixed in #993. However, I think that the latest versions should fix the "if you delete C-u mapping it scrolls 100 lines" issue?
